src/fact_base.py

```diff
@@ -3,10 +3,24 @@
 import signal
 from time import sleep
 
-import psutil
-
-from helperFunctions.program_setup import program_setup
-from statistic.work_load import WorkLoadStatistic
+try:
+    import psutil
+    import psycopg2  # pylint: disable=unused-import  # noqa: F401  # new dependency of FACT>=4.0
+
+    from helperFunctions.program_setup import program_setup
+    from statistic.work_load import WorkLoadStatistic
+except ImportError:
+    import sys
+    logging.error(
+        'Could not load dependencies. Please make sure that you have installed FACT correctly '
+        '(see INSTALL.md for more information). If you recently updated FACT, you may want to rerun the installation.'
+    )
+    logging.warning(
+        'The database of FACT switched from MongoDB to PostgreSQL with the release of FACT 4.0. '
+        'For instructions on how to upgrade FACT and how to migrate your database see '
+        'https://fkie-cad.github.io/FACT_core/migration.html'
+    )
+    sys.exit(1)
```

Comment:
You should be able to replace this by a raise of the original error. Then you can keep the syntax in the dependent files and loose the unnecessary `sys.exit` here.

Reply:
added the `raise` and changed the `logging.error` to a `logging.exception` so that we still know what caused the error